src/main/java/org/fogbowcloud/manager/core/plugins/compute/openstack/OpenStackNovaV2ComputePlugin.java

```diff
@@ -0,0 +1,438 @@
+import org.fogbowcloud.manager.core.models.orders.instances.ComputeOrderInstance;
+import org.fogbowcloud.manager.core.models.token.Token;
+import org.fogbowcloud.manager.core.plugins.compute.ComputePlugin;
+import org.fogbowcloud.manager.core.utils.HttpRequestUtil;
+import org.json.JSONArray;
+import org.json.JSONException;
+import org.json.JSONObject;
+
+import java.nio.charset.StandardCharsets;
+import java.util.*;
+
+public class OpenStackNovaV2ComputePlugin implements ComputePlugin {
+
+    private static final String ID_JSON_FIELD = "id";
+    private static final String NAME_JSON_FIELD = "name";
+    private static final String SERVER_JSON_FIELD = "server";
+    private static final String FLAVOR_JSON_FIELD = "flavorRef";
+    private static final String IMAGE_JSON_FIELD = "imageRef";
+    private static final String USER_DATA_JSON_FIELD = "user_data";
+    private static final String NETWORK_JSON_FIELD = "networks";
+    private static final String DISK_JSON_FIELD = "disk";
+    private static final String VCPU_JSON_FIELD = "vcpus";
+    private static final String MEMORY_JSON_FIELD = "ram";
+    private static final String FLAVOR_JSON_OBJECT = "flavor";
+    private static final String KEY_JSON_FIELD = "key_name";
+    private static final String PUBLIC_KEY_JSON_FIELD = "public_key";
+    private static final String KEYPAIR_JSON_FIELD = "keypair";
+    private static final String UUID_JSON_FIELD = "uuid";
+    private static final String FOGBOW_INSTANCE_NAME = "fogbow-instance-";
+    private static final String TENANT_ID = "tenantId";
+    private static final String SERVERS = "/servers";
+    private static final String SUFFIX_ENDPOINT_KEYPAIRS = "/os-keypairs";
+    private static final String SUFFIX_ENDPOINT_FLAVORS = "/flavors";
+    private static final String COMPUTE_NOVAV2_URL_KEY = "compute_novav2_url";
+    private static final String NO_VALID_HOST_WAS_FOUND = "No valid host was found";
+    private final String COMPUTE_V2_API_ENDPOINT = "/v2/";
+
+    private static final String COMPUTE_NOVAV2_NETWORK_KEY = "compute_novav2_network_id";
+
+    private static final Logger LOGGER = Logger.getLogger(OpenStackNovaV2ComputePlugin.class);
+
+    private List<Flavor> flavors;
+    private Properties properties;
+    private HttpClient client;
+
+    public OpenStackNovaV2ComputePlugin(Properties properties) {
+        this.flavors = new ArrayList<>();
+        this.properties = properties;
+        this.initClient();
+    }
+
+    public String requestInstance(ComputeOrder computeOrder, String imageId) throws RequestException {
+        LOGGER.debug("Requesting instance with token=" + computeOrder.getLocalToken());
+
+        Token localToken = computeOrder.getLocalToken();
+        String flavorId = updateAndFindSmallestFlavor(computeOrder).getId();
+        String tenantId = getTenantId(localToken);
+        String networkId = getNetworkId();
+        String userData = computeOrder.getUserData().getContent();
```

Comment:
These methods chaining calls are bad for debug.

Reply:
Fixed!